packages/core/src/browser/tree/tree-model.ts

```diff
@@ -260,7 +260,7 @@ export class TreeModelImpl implements TreeModel, SelectionProvider<ReadonlyArray
         return undefined;
     }
 
-    protected *getExpansionCandidates(raw?: Readonly<TreeNode>): IterableIterator<TreeNode | undefined> {
+    protected *getExpansionCandidates(raw?: Readonly<TreeNode>): Generator<TreeNode | undefined, void, undefined> {
```

Comment:
This suggests that this change is not necessarily desirable: that is, the `Generator` type is more specific than the `IterableIterator`, and rewriting `IterableIterator<T>` as `Generator<T>` as is done in most places in this PR is potentially breaking: `Array<T>.values()` (or `*yield someArray<T>`) satisfies `IterableIterator<T>` but not `Generator<T>`. See [this playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBApgDwIYFsAOAbOMC8MxwDuMAggE5lICeAPGAK4oBGcZAfABQCUA3AFB9QkWADMkASwwQAXDADicApSggydRi3a54ydFgB0ANyQZ6cCN36Dw0GBHrBgcOABMZMAJJRWSJli8+KmoMzKxs2oiomHBGJmYWvHwwMEA) for a simple demonstration.

Reply:
The issue with your playground is that we would never try to assign `IterableIterator<T>` to a `Generator<...>` type.

[But `Generator` is assignable to `IterableIterator`](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAKgOYFMzoE4EMpzYAUAlAFyIDimO+hAPGCALYBGOAfIgN4BQiAxAE8Y6ADYATRAEZeAXwDcvXhAQBnKIjUgIEdOglqKASSi1WY9KdoFsjFu2xcAvIgxY8t0kqA).

I guess the most annoying bit from `Generator` is how they typed `TNext = unknown` as it makes it incompatible with `IterableIterator` as it has `TNext = undefined`... But occurrences of this problem are low, and can be fixed by specifying the templates types.

Reply:
> [But `Generator` is assignable to `IterableIterator`](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAKgOYFMzoE4EMpzYAUAlAFyIDimO+hAPGCALYBGOAfIgN4BQiAxAE8Y6ADYATRAEZeAXwDcvXhAQBnKIjUgIEdOglqKASSi1WY9KdoFsjFu2xcAvIgxY8t0kqA).

Exactly - so we should stick with `IterableIterator` :-). I do agree that it's a pain that they typed them in a way that means they aren't compatible by default when they didn't really have to do that, but given that they have, we're safer sticking with `IterableIterator` where we've been using it, though I'm not opposed to using `Generator` where appropriate in the future.

Reply:
> though I'm not opposed to using Generator where appropriate in the future.

Will do then :)

I'd still prefer to changes previous occurrences, but this isn't really a hill I'm willing to die on.